src/errors.ts

```diff
@@ -312,3 +312,49 @@ export class ProtocolError extends FaunaError {
     this.httpStatus = error.httpStatus;
   }
 }
+
+const QUERY_CHECK_FAILURE_CODES = [
+  "invalid_function_definition",
+  "invalid_identifier",
+  "invalid_query",
+  "invalid_syntax",
+  "invalid_type",
+];
+
+export const getServiceError = (
```

Comment:
Let's write a unit test for this since it is now a public method.

Reply:
Good idea! Will do.